python/mxnet/gluon/block.py

```diff
@@ -520,11 +520,11 @@ def _infer_attrs(self, infer_fn, attr, *args):
                 **{i.name: getattr(j, attr) for i, j in zip(inputs, args)})
             if arg_attrs is None:
                 raise ValueError(w[0].message)
-            sdict = {i: j for i, j in zip(out.list_arguments(), arg_attrs)}
-            sdict.update({name : attr for name, attr in \
-                 zip(out.list_auxiliary_states(), aux_attrs)})
-            for i in self.collect_params().values():
-                setattr(i, attr, sdict[i.name])
+        sdict = {i: j for i, j in zip(out.list_arguments(), arg_attrs)}
```

Comment:
arg_attrs and aux_attrs will be none at this point since they obtain values within warning block and you have moved this chunk of code outside warning block. Wouldn't that be the case?

Reply:
no. that's not how python variable scope works